Feature/dynamic cdn enhancements

aaronbrazell opened the pull request.

aaronbrazell committed:
v 0.2.1

aaronbrazell committed:
min stability

aaronbrazell committed:
Add support for WP 4.4 image srcsets

aaronbrazell committed:
Always run the srcset operation

ericmann committed:
Merge pull request #6 from technosailor/feature/wp-image-srcset-support

Add support for WP 4.4 image srcsets

ericmann committed:
Bump versions and clean up some documentation

ericmann committed:
Action/Filter documentation

ericmann committed:
Bump readme

aaronbrazell committed:
merge conflicts

aaronbrazell committed:
formatting

aaronbrazell committed:
Don't apply srcset replacement in the admin and use the already defined filter for http/s scheme

aaronbrazell committed:
inline docs

ericmann committed:
Merge pull request #8 from technosailor/feature/srcsets

Feature/srcsets

ericmann committed:
Begin wiring in test suite

ericmann committed:
Reorganize things

ericmann committed:
Begin refactoring the multiton class into a namespace for easier testing

ericmann committed:
Start breaking things up a bit

ericmann committed:
Flesh out some serious tests

ericmann committed:
Further code migration

ericmann committed:
Add 5.4 to the test matrix

ericmann committed:
Fix src_dir

ericmann committed:
Collect DomainManager instances

ericmann committed:
Remove an ugly pseudoglobal

ericmann committed:
Bump things over

ericmann committed:
Add ability to select 'last' manager initialized

ericmann committed:
Utilize the new ::last() lookup

ericmann committed:
Add another action expectation

ericmann committed:
Version bump

ericmann committed:
Merge unit test RC to master

ericmann committed:
It seems Coveralls has changed instructions ...

ericmann committed:
Remove deprecated file that my other projects still have for some reason

aaronbrazell committed:
Update to 0.4

aaronbrazell committed:
Port original changes

aaronbrazell committed:
Bump version

aaronbrazell committed:
increment CDN version in composer.json

aaronbrazell committed:
Support multiple domain types

aaronbrazell committed:
WIP

aaronbrazell committed:
Effectively support two separate domains

aaronbrazell committed:
remove debug code and fix most tests

aaronbrazell committed:
Fix last test

pdufour commented on includes/functions/core.php:
```diff
@@ -0,0 +1,298 @@
+use EAMann\Dynamic_CDN\DomainManager;
+
+// @todo if this was a class, we could self contain this variable. It's needed in multiple functions so we need to put it in the namespaced global scope
+$context = 'uploads';
```
i think we should name this variable something else if it's going to be a global. like with a prefix.

pdufour replied:
`dyncd_context` maybe?

pdufour commented on tests/phpunit/DomainManagerTests.php:
```diff
@@ -0,0 +1,121 @@
+
+	public function test_has_domains() {
+		$manager = DomainManager( 'http://test.com' );
+		$this->assertFalse( $manager->has_domains( 'uploads' ) );
```
wonder why this test wasn't working? isn't `uploads` the default value? or are you just adding this to be explicit.

aaronbrazell committed:
uniquify the global variable

aaronbrazell committed:
Update core.php

s/DYNCDN_ASSET_DOMAIN/DYNCDN_ASSET_DOMAINS/

Commit:
ensure port doesn't leak into replacement

Commit:
fix typo and designate host portion of url

aaronbrazell committed:
make sure wp-content/uploads is present in URLs for uploads

aaronbrazell committed:
use wp_upload_dir()

pdufour committed:
Revert to master functionality for getting the search & replacement domain

pdufour committed:
Merge branch 'feature/dynamic-cdn-enhancements' of github.com:britco/dynamic-cdn into feature/dynamic-cdn-enhancements

pdufour committed:
Bug fixes & optimizations

- Fix srcset not working
- Change regex to more performant regex (see https://regex101.com/r/zP8sC9/1)
- Re-escape slashes in filter_cb if they were previously escaped

pdufour committed:
Update #2

pdufour committed:
Remove references to domain & just use last

pdufour committed:
add back in ^/ part

pdufour committed:
Add back protocol

pdufour committed:
Use protocol if one exists

pdufour committed:
re-add domain but use it instead of just calling ::last()

pdufour committed:
Re-add comment

pdufour committed:
move filter to function so it's applied everywhere

pdufour committed:
include comment about protocol

pdufour committed:
cleanup

aaronbrazell committed:
Don't process content images twice

aaronbrazell committed:
Merge branch 'feature/dynamic-cdn-enhancements' of github.com:britco/dynamic-cdn into feature/dynamic-cdn-enhancements

pdufour committed:
edit comment

pdufour committed:
If filtered on uploads, change the context

pdufour committed:
add unit test for escaped content
